Instagram/View Controllers/LoginViewController.m

```diff
@@ -0,0 +1,116 @@
+//
+//  LoginViewController.m
+//  Instagram
+//
+//  Created by Mai Ngo on 6/27/22.
+//
+
+#import "LoginViewController.h"
+#import "Parse/Parse.h"
+#import "Foundation/Foundation.h"
+
+@interface LoginViewController ()
+@property (weak, nonatomic) IBOutlet UITextField *usernameField;
+@property (weak, nonatomic) IBOutlet UITextField *passwordField;
+
+@end
+
+NSString *const loginSegue = @"loginSegue";
+
+@implementation LoginViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+}
+- (IBAction)onTapRegister:(id)sender {
+    [self registerUser];
+}
+- (IBAction)onTapLogin:(id)sender {
+    dispatch_semaphore_t sema = dispatch_semaphore_create(0);
```

Comment:
I know this isn't relevant to the IG project since this code shouldn't be here at all, but using a semaphore will cause the thread to block until this method is complete, which is not ideal. Someone who has slow internet will see a frozen screen. Typically the way to do this to show some kind of loading state while the network request is in progress, then use the completion handler to transition to the next step.